SpinRite support

miki725 opened the pull request:
Adding [SpinRite](https://www.grc.com/sr/spinrite.htm) support. Script copied from [here](https://www.linuxquestions.org/questions/linux-newbie-8/need-a-way-to-run-spinrite-6-via-grub2-can-do-either-iso-or-image-file-4175546458/) and modified to work with the scripts structure in this repo. It seems to be working well. Successfully running SpinRite.

aguslr commented:
Hi @miki725! Thanks for the PR.

I see that, in your commit, you have hard-coded a disk and partitoin (`hd0,msdos5`) as *root*. This will not match all environments and *root* is already set to the USB device. Also, your if-clause is used to set *rootuuid* which is also set.

This ISO is loaded using MEMDISK so I believe something more standard should work. Have you tried this:

```
for isofile in $isopath/SpinRite*.iso; do
  if [ -e "$isofile" ]; then
    regexp --set=isoname "$isopath/(.*)" "$isofile"
    menuentry "$isoname (memdisk)" "$isofile" {
      iso_path="$2"
      bootoptions="iso raw"
      linux16 $prefix/memdisk $bootoptions
      initrd16 $iso_path
    }
  fi
done
```

This is what we use for other ISO files loaded with MEMDISK (e.g. Hiren's BootCD, UBCD, etc.).

Also, could you please rename the configuration file to `generic.cfg` so that it matches the rest of the repository files?

Thanks!

miki725 commented:
@aguslr thanks for the quick response.

initially I copied some other other config which did not work so googled for alternatives. anyway, your script works like a charm and looks way simpler. thanks!

replaced script and renamed to `generic.cfg`

let me know if anything else needs to change.
